### PR TITLE
MM-38745 feat: Add dont clear option for DND

### PR DESCRIPTION
**Pull request by tanmaythole**
#### Summary

This PR adds the `Don't Clear` option at DND dropdown.

#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/19134

#### Release Note

```release-note
Added a 'Don't Clear' option for Do Not Disturb
```

### CODE REVIEW

**Review by matthewbirtch — APPROVED** [review]
Looks great to me @tanmaythole. Thanks for the work on this!

File: webapp/channels/src/components/widgets/menu/menu_items/submenu_item.tsx [hunk 1]
```diff
@@ -112,8 +112,13 @@ export default class SubMenuItem extends React.PureComponent<Props, State> {
             } else if (action) { // leaf node in the tree handles action only
                 action(postId);
             }
-        } else if (event.currentTarget.id === id && action) {
-            action(postId);
+        } else {
+            const shouldCallAction =
```
Comment (hmhealey) [inline webapp/channels/src/components/widgets/menu/menu_items/submenu_item.tsx:116]: Are these changes still needed now that the duplicate `onClick` has been removed from below? I can't 100% remember, but based on my comment [here](https://github.com/mattermost/mattermost/pull/25054#pullrequestreview-1816795377), I think the changes to this method can be removed

File: webapp/channels/src/components/widgets/menu/menu_items/submenu_item.tsx [hunk 2]
```diff
@@ -233,7 +238,8 @@ export default class SubMenuItem extends React.PureComponent<Props, State> {
                     aria-label={ariaLabel}
                     onMouseEnter={this.show}
                     onMouseLeave={this.hide}
-                    onClick={this.onClick}
+
+                    // onClick={this.onClick}
```
Comment (hmhealey) [inline webapp/channels/src/components/widgets/menu/menu_items/submenu_item.tsx:242]: Could you remove this?

**Review by hmhealey — CHANGES_REQUESTED** [review]
Thanks for resubmitting this! I really appreciate you sticking with this.

File: webapp/channels/src/components/widgets/menu/menu_items/submenu_item.tsx (hunk 1 above)
Comment (tanmaythole) [reply webapp/channels/src/components/widgets/menu/menu_items/submenu_item.tsx:116]: Yes, it still requires as we also need to handle keydown

File: webapp/channels/src/components/widgets/menu/menu_items/submenu_item.tsx (hunk 2 above)
Comment (tanmaythole) [reply webapp/channels/src/components/widgets/menu/menu_items/submenu_item.tsx]: Done 👍🏼

**Review by hmhealey — APPROVED** [review]

**Review by yasserfaraazkhan — APPROVED** [review]

**Review by calebroseland — APPROVED** [review]

**Comment by hmhealey** [comment]
Thanks again for the PR, @tanmaythole! I'm excited to not trip myself up any more by clicking on "Do not disturb" and having the menu close without anything happening 😁

**Pull request merged**